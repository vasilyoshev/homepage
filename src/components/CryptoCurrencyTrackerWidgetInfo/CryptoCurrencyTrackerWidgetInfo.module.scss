.mainCard {
  width: 320px;
  height: 180px;
  padding: 3px;
  margin-bottom: 5px;
  background-color: rgb(235, 229, 229);
  border: 1px solid #808080;
}

.coinName {
  font-size: 22px;
}

.icon {
  width: 35px;
  height: 35px;
  padding-left: 5px;
}

.coinPrice {
  display: inline-block;
  padding: 3px;
  margin: 8px 0 0;
  font-size: 20px;
  border: 1px solid #808080;
  border-radius: 5px;
}

.marketcap {
  display: inline-block;
  margin-top: 5px;
}

.coinPriceChange {
  display: inline-block;
  margin-top: 8px;
  font-size: 20px;
}

.coinPercentRed {
  display: inline;
  padding: 3px;
  margin: 5px 0 10px 5px;
  font-size: 22px;
  color: #f00606;
  border: 2px solid #f00606;
  border-radius: 5px;
}

.coinPercentGreen {
  display: inline;
  padding: 3px;
  margin: 5px 0 10px 5px;
  font-size: 22px;
  color: #0fbd0f;
  border: 2px solid #0fbd0f;
  border-radius: 5px;
}
