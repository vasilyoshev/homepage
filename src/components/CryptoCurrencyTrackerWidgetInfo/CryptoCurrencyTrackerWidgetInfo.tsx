import React from 'react';
import { CryptoCurrencyTrackerWidgetInfoProps } from 'interfaces';
import { Card } from '@material-ui/core';
import styles from './CryptoCurrencyTrackerWidgetInfo.module.scss';

export const CryptoCurrencyTrackerWidgetInfo: React.FC<CryptoCurrencyTrackerWidgetInfoProps> = ({ coins }) => {
  return (
    <>
      {!!coins.length && coins.map((coin) => (
        <Card className={styles.coinCard} key={coin.id}>
          <div className={styles.coinName}>
            {coin.name}
            <img
              className={styles.icon}
              src={coin.image}
              alt="crypto" />
          </div>
          <div className={styles.coinPrice}>Price: ${coin.current_price.toFixed(2)}</div>
          <div className={styles.coinMarketcap}>Market Cap: ${coin.market_cap.toFixed(2)}</div>
          <div className={styles.coinPriceChange}>
              Price change 24h:
            {coin.price_change_percentage_24h < 0 ? (
              <p className={styles.coinPercentRed}>{coin.price_change_percentage_24h.toFixed(2)}%</p>
            ) : (
              <p className={styles.coinPercentGreen}>+{coin.price_change_percentage_24h.toFixed(2)}%</p>
            )}
          </div>
        </Card>
      ))}
    </>
  );
};
