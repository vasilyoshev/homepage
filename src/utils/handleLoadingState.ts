import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
