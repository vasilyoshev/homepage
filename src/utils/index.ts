export * from './handleLoadingState';
export * from './testUtils';
