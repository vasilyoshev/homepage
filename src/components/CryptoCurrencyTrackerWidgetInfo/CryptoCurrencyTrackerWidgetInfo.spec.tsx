import { render } from 'utils';
import { CryptoCurrencyTrackerWidgetInfo } from './CryptoCurrencyTrackerWidgetInfo';

describe('CryptoCurrencyTrackerWidgetInfo', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<CryptoCurrencyTrackerWidgetInfo coins={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
