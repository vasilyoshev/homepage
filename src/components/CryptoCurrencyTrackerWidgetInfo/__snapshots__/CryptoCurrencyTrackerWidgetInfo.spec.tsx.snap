// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoCurrencyTrackerWidgetInfo renders successfully 1`] = `<DocumentFragment />`;
